### PR TITLE
send a control packet as bandwidth probe

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

File: libraries/networking/src/udt/SendQueue.cpp [hunk 1]
```diff
@@ -463,6 +463,12 @@ bool SendQueue::maybeSendNewPacket() {
         // do we have a second in a pair to send as well?
         if (secondPacket) {
             sendNewPacketAndAddToSentList(move(secondPacket), getNextSequenceNumber());
+        } else {
```
Comment (Atlante45) [inline libraries/networking/src/udt/SendQueue.cpp]: `else if (((uint32_t) nextNumber & 0xF) == 1)`

Comment (birarda) [inline libraries/networking/src/udt/SendQueue.cpp]: Oh woof, good call. Fixing

**Comment by Atlante45** [comment]
Just a though but as an improvement in the future we might want to add a Sequence number to tail control packet to discard the messed up pairs.

(ie. tail control packet arrives after next packet)

**Comment by Atlante45** [comment]
Actually I think that's a case we should address now. We don't need a Sequence number to fix it.

We can do something like if the last packet received doesn't have a sequence number which is equal to 0 modulo one, discard.

**Pull request merged**